hail/python/hail/expr/types.py

```diff
@@ -1424,7 +1424,25 @@ def _parsable_string(self):
         return "Call"
 
     def _convert_from_json(self, x):
-        return hl.Call._from_java(hl.Call._call_jobject().parse(x))
+        if x == '-':
+            return hl.Call([])
+        if x == '|-':
+            return hl.Call([], phased=True)
+        if x[0] == '|':
+            return hl.Call([int(x[1:])], phased=True)
+
+        n = len(x)
+        i = 0
+        while i < n:
+            c = x[i]
+            if c in '|/':
+                break
+            i += 1
+
+        if i == n:
+            return hl.Call([int(x)])
+
+        return hl.Call([int(x[0:i]), int(x[i+1:])], phased=(c == '|'))
```

Comment:
This used to throw nice, user-legible errors on invalid calls or if there were more than 2 alleles. Is this method strictly internal enough that we don't need to care about those?

Reply:
I don't think this is validating user input.  It should only be handling data coming from the JVM/service.  `parse` is still used for the String Call constructor in the expression langauge.  I don't believe we have a Python-side parse to literal Call that's user visible.